devito/propagator.py

```diff
@@ -47,11 +50,12 @@ class Propagator(object):
                            Set cache_blocking to AutoTuner instance, to use auto tuned
                            tune block sizes
     """
-    def __init__(self, name, nt, shape, stencils, factorized=None, spc_border=0,
-                 time_order=0, time_dim=None, space_dims=None, dtype=np.float32,
-                 forward=True, compiler=None, profile=False,
+    def __init__(self, name, nt, shape, stencils=None, loop_body=None, factorized=None,
```

Comment:
Apart from the reason detailed above, I think we should keep a door open for someone who knows the exact C statement they want to add to the generated code and where they want to add it. In that sense, `stencils` is the door for someone who wants to give us a symbolic expression, and `loop_body` is the door for someone who would rather give us the C statement directly.